src/firehose.js

```diff
@@ -40,7 +40,9 @@ const FireHoser = class FireHoser extends IFireHoser {
       },
     };
 
-    return this.firehose.putRecord(params).promise();
+    return this.firehose.putRecord(params).promise().catch((err) => {
+      console.error(err)
+    });
```

Comment:
hey sorry for the back and forth but I took a look at some other transports to see how they handle errors, apparently the paradigm is now something like [this](https://github.com/winstonjs/winston-riak/blob/master/lib/winston-riak.js#L66-L73):

```javascript
    if (err) {
      self.emit('error', err);
      return callback(err, false);
    }

    callback(null, true);
    self.emit('logged');
```

So the equivalent here would be to change the log function [here](https://github.com/pkallos/winston-firehose/blob/master/src/firehose.js#L67-L77)

to something like this:

```javascript
  log(info, callback) {
    if (callback) {
      setImmediate(callback);
    }
    let message = info[MESSAGE];
    if (this.formatter) {
      message = Object.assign({ timestamp: (new Date()).toISOString() }, info);
      message = this.formatter(message);
    }
    return this.firehoser.send(message)
        .then(() => 
           self.emit('logged');
           callback(null, true);
         }).catch((err) => {
           self.emit('error', err);
           callback(err, false);
    });
  }
```

Sorry for the additional trouble but if you have a stream setup to test on would you mind patching that in and seeing if it behaves as expected and fixes the silent failure to boot?

Reply:
No worries! I will set that up and post updates when I have them.

Reply:
@thomasstep any progress on testing this out?

Reply:
@pkallos Unfortunately not much. I have not had time to swivel back to this since the original work around is working. I did try to make that change, but there were still errors being thrown. I'm guessing it had to do with me not setting the `exitOnError`, `handleExceptions`, or `handleRejections` properties on the logger, but I'm not 100% sure.